app/src/main/java/org/dharmaseed/android/TalkRepository.java

```diff
@@ -31,37 +31,39 @@ public Cursor getTalkAdapterData(
             boolean isDownloaded
     )
     {
-        List<String> columns = new ArrayList<>();
-        columns.add(DBManager.C.Talk.TABLE_NAME    + "." + DBManager.C.Talk.ID);
-        columns.add(DBManager.C.Talk.TABLE_NAME    + "." + DBManager.C.Talk.TITLE);
-        columns.add(DBManager.C.Talk.TABLE_NAME    + "." + DBManager.C.Talk.TEACHER_ID);
-        columns.add(DBManager.C.Teacher.TABLE_NAME + "." + DBManager.C.Teacher.NAME);
-        return getTalks(columns, searchTerms, isStarred, isDownloaded);
+
+        return getTalks(searchTerms, isStarred, isDownloaded, "");
     }
 
     /**
-     * @param columns the columns to select or null if all
      * @param searchTerms the term to filter by or null if none
      * @param isStarred whether the talk is starred
      * @param isDownloaded whether the talk is downloaded
+     * @param extraWhere an extra where clause that can be used to filter on a specific teacher/center id
      * @return every talk in the db filtered by search term, is starred, and is downloaded
      */
     private Cursor getTalks(
-           List<String> columns,
            List<String> searchTerms,
            boolean isStarred,
-           boolean isDownloaded
+           boolean isDownloaded,
+           String extraWhere
     )
     {
+        List<String> columns = new ArrayList<>();
+        columns.add(DBManager.C.Talk.TABLE_NAME    + "." + DBManager.C.Talk.TITLE);
+        columns.add(DBManager.C.Talk.TABLE_NAME    + "." + DBManager.C.Talk.TEACHER_ID);
+        columns.add(DBManager.C.Teacher.TABLE_NAME + "." + DBManager.C.Teacher.NAME);
+        columns.add(DBManager.C.Center.TABLE_NAME    + "." + DBManager.C.Center.NAME);
+        columns.add(DBManager.C.Talk.TABLE_NAME    + "." + DBManager.C.Talk.DURATION_IN_MINUTES);
+        columns.add(DBManager.C.Talk.TABLE_NAME    + "." + DBManager.C.Talk.RECORDING_DATE);
+
         String queryColumns = "";
-        if (columns == null || columns.size() == 0)
```

Comment:
I couldn't find any place where this branch was used, but if I missed something please do let me know. :)

Reply:
You're right, it's not currently used. At first I was thinking we should keep it, but actually I like removing it. It's likely that the caller will _never_ need to `SELECT *` and, if they do, we should have them pass in every column they need anyway. This way no one will accidentally `SELECT *`. But we should still perform the check to prevent generating a query like `SELECT FROM talks` - instead of `queryColumns = "*"`, what do you think we should do instead? Just return a null cursor?

This will require changing a test.

Reply:
Cool, yeah, I'd say logging an error and returning a null cursor if the caller passes in an empty list would be fine. I'll go ahead and do that.  Given that the lists of columns are almost certain to be static and known at compile time, I think it'll be pretty obvious when developing new queries if there is a logical problem anywhere.